debugtools/DDR_VM/src/com/ibm/j9ddr/tools/ddrinteractive/CacheMemory.java

```diff
@@ -0,0 +1,155 @@
+ * General Public License, version 2 with the GNU Classpath                                                                                                                  
+ * Exception [1] and GNU General Public License, version 2 with the                                                                                                          
+ * OpenJDK Assembly Exception [2].                                                                                                                                           
+ *                                                                                                                                                                           
+ * [1] https://www.gnu.org/software/classpath/license.html                                                                                                                   
+ * [2] http://openjdk.java.net/legal/assembly-exception.html                                                                                                                 
+ *                                                                                                                                                                           
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception                                                                                                                                                                                                                
+ *******************************************************************************/
+package com.ibm.j9ddr.tools.ddrinteractive;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.nio.ByteOrder;
+import java.util.Properties;
+
+import com.ibm.j9ddr.corereaders.memory.AbstractMemory;
+import com.ibm.j9ddr.corereaders.memory.IAddressSpace;
+import com.ibm.j9ddr.corereaders.Platform;
+import com.ibm.j9ddr.corereaders.memory.IModule;
+import com.ibm.j9ddr.corereaders.memory.IProcess;
+import com.ibm.j9ddr.corereaders.memory.IMemorySource;
+import com.ibm.j9ddr.corereaders.memory.MemoryFault;
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.DataUnavailableException;
+import com.ibm.j9ddr.corereaders.osthread.IOSThread;
+
+/**
+ * Object representing a single live
+ * process model where a section of
+ * memory in a the process can be represented 
+ * in a byte buffer as its source 
+ * @see com.ibm.j9ddr.tools.ddrinteractive.CacheMemorySource
+ *
+ * Can be used for inspection of the
+ * scc, without the need of a core file
+ *
+ * Behaviour of the model is not defined
+ * for the case where the underlying source's
+ * ByteBuffer is modified (address or capacity)
+ * during use.
+ *
+ * @author knewbury01
+ * 
+ */
+
+public class CacheMemory extends AbstractMemory implements IProcess
```

Comment:
Please provide a more complete example of how this class is intended to be used. It's unclear how the implementation is sufficient (e.g. returning null from several methods).

Reply:
added complete, generalized example of how it can be used, in the class javadoc as a code snip.
should demonstrate a few other things that can be done with it, 
scc specific, 
hopefully sufficiently clear!